modules/managers/galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.selenium.manager/src/main/java/dev/galasa/selenium/internal/ChromeOptionsImpl.java

```diff
@@ -59,9 +59,15 @@ public void setAcceptInsecureCerts(boolean bool) {
         options.setAcceptInsecureCerts(bool);
     }
 
+    /**
+     * @deprecated Use {@link IChromeOptions#addArguments(arguments)} instead.
+     */
+    @Deprecated(forRemoval = true, since = "0.39.0")
     @Override
-    public void setHeadless(boolean bool) {
-        options.setHeadless(bool);
+    public void setHeadless(boolean isHeadless) {
+        if (isHeadless) {
+            options.addArguments(List.of("--headless"));
+        }
```

Comment:
`setHeadless` no longer exists in Selenium 4 - see https://www.selenium.dev/blog/2023/headless-is-going-away/.

Added a temporary implementation to keep this method available for now, but Selenium doesn't expose any methods to remove arguments from options, so calling `setHeadless(false)` won't do anything.

One way to remove arguments is to create a new `options` object, but this means losing any config that was done to the object previously.